developer-experience/readme.md

```diff
@@ -0,0 +1,130 @@
+- Minimize the amount of time spent manual setup and configuration of tooling
+- Minimize regressions and new defects by making end-to-end testing easy
+
+## Impact
+
+Developer experience can have a significant impact on the efficiency of the day to day execution of the team. A positive experience can pay dividends throughout the lifetime of the project; especially as new developers join the team.
+
+- Increased Velocity - Team spends less time on non-value-add activities such as dev/local environment setup, waiting on remote environments to test, and rework (fixing defects).
+- Improved Quality - When its easy to debug and test, developers will do more of it. This will translate to fewer defects being introduced.
+- Easier Onboarding & Adoption - When dev essential tasks are automated, there is less documentation to write and, subsequently, less to read to get started!
+
+## Measures
+
+### Time to First Debug (aka F5 Contract)
+
+Assuming a laptop/pc that has never run the solution, how long does it take to start debugging the solution end-to-end.
+
+### Time To First Commit
+
+How long does it take to make a change that can be verified/tested locally. A locally verified/tested change is one that passes test cases without introducing regression or breaking changes.
+
+## Participation
+
+The entire development team should take responsibility in ensuring positive
+
+## Facilitation Guidance
+
+The following outline several strategies that can be adopted to promote a positive developer experience.
+
+### Establish Hotkeys
+
+Assign hotkeys to each of the essential tasks.
+
+[`TODO: Suggest hotkeys for mac`]
+
+| Task                 | Windows      |
+| -------------------- | ------------ |
+| Build                | CTRL+SHIFT+B |
+| Test                 | CTRL+R,T     |
+| Start With Debugging | F5           |
+
+### The F5 Contract
+
+The F5 contract aims for the ability to run the end-to-end solution with the following steps.
+
+1. Clone - git clone [`my-repo-url-here`]
+2. Configure - set any configuration values that need to be unique to the individual (i.e. update a .env file)
+3. Press F5 - launch the solution with debugging attached.
+
+Most IDE's have some form of a task runner that can be used to automate the build, execute, and attach steps. Try to leverage these such that the steps can all be run with as few manual steps as possible.
+
+### Pay Attention to the Struggles
+
+Ask the team to surface the areas they spend significant time in that they consider non-value-add activities. For example, do they need to deploy their changes to an environment off their laptop before they can validate if what they did worked. Rather than debugging locally, do they have to do this repetitively to get to a working solution? Does this take several minutes each iteration? Does this block other developers due to the contention on the environment?

```

Comment:
Doubling down on the ownership aspect... the owner of project devex should proactively seek out feedback on how the current approach is working, where it can improve, soliciting best practices from other teams, and then incorporating stories/tasks for improvement items into sprint planning.

Retros are of course a way to circulate this information, but we shouldn't merely wait for problems to bubble up. Let's promote a mindset of seeking them out and eliminating them, actively.

Reply:
Great point, will reword this to assign ownership of this aspect to the DevEx Champion